sourmash_lib/sbt_storage.py

```diff
@@ -0,0 +1,152 @@
+from __future__ import print_function, unicode_literals, division
+
+import abc
+from io import BytesIO
+import os
+import tarfile
+
+
+class Storage(abc.ABCMeta(str('ABC'), (object,), {'__slots__': ()})):
```

Comment:
The `Storage` class defines what methods need to be implemented by a storage. Also defines generic `__enter__` and `__exit__` methods to make it into a context manager.